sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventHubSharedKeyCredential.java

```diff
@@ -82,11 +98,21 @@ public EventHubSharedAccessKeyCredential(String policyName, String sharedAccessK
             throw new IllegalArgumentException("'tokenTimeToLive' has to positive and in the order-of seconds");
         }
 
-        hmac = Mac.getInstance(HASH_ALGORITHM);
+        try {
+            hmac = Mac.getInstance(HASH_ALGORITHM);
+        } catch (NoSuchAlgorithmException e) {
+            throw logger.logExceptionAsError(new AzureException(
```

Comment:
Is `AzureException` the appropriate exception to be thrown here?

Reply:
Any suggestions? Not sure what would apply.